cmctl: 1.8.2 -> 1.9.1

JoshVanL opened the pull request:
###### Description of changes

https://cert-manager.io

Updates cmctl to version `v1.9.1`.


Adds `joshvanl` as a maintainer, and adds `joshvanl` as a maintainer for `cmctl`.

To inspect upstream changes

- [Release on GitHub](https://github.com/cert-manager/cert-manager/releases/tag/v1.9.1)
-  [Compare changes on GitHub](https://github.com/cert-manager/cert-manager/compare/v1.8.0...v1.9.1)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
~- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))~
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

JoshVanL committed:
cmctl: 1.8.2 -> 1.9.1

Updates cmctl to v1.9.1

Adds `joshvanl` as cmctl maintainer.

Signed-off-by: joshvanl <me@joshvanl.dev>

superherointj commented on pkgs/applications/networking/cluster/cmctl/default.nix:
```diff
@@ -2,33 +2,49 @@
-    repo = "cert-manager";
-    rev = "v${version}";
-    sha256 = "sha256-sfC1acnCrcQ4A1tXXcjh47Af6xeJqjdGXy0gK21ZSFg=";
+    repo  = "cert-manager";
```
```suggestion
    repo = "cert-manager";
    rev = "v${version}";
    hash = "sha256-Z1aJ18X4mfJPlCPBC7QgfdX5Tk4+PK8mYoJZhGwz9ec=";
```

JoshVanL commented:
@ofek @superherointj anything left for me to do?

superherointj commented:
> @ofek @superherointj anything left for me to do?

Yes.

ofek commented:
Why were we pinged? 🤔

superherointj commented:
> Why were we pinged? thinking

Because @JoshVanL mentioned you?

ofek commented:
Right, why?

JoshVanL commented:
Sorry! Looks like I fat fingered the @ 🙂

JoshVanL committed:
maintainers: add joshvanl

Signed-off-by: joshvanl <me@joshvanl.dev>

JoshVanL commented:
I've gone ahead and added myself to the maintainers list and added myself as a maintainer to the cmctl package.

I personally use nixos and am a maintainer of [cert-manager](https://github.com/cert-manager/cert-manager).

Packaging and releasing cert-manager is part of my day job :slightly_smiling_face:

superherointj commented:
> I've gone ahead and added myself to the maintainers list and added myself as a maintainer to the cmctl package.
> I personally use nixos and am a maintainer of [cert-manager](https://github.com/cert-manager/cert-manager).
> Packaging and releasing cert-manager is part of my day job slightly_smiling_face

Great.

I gave you feedback 4 times. I haven't heard from you on any of them.
For example, you are using Go style indentation instead of Nix indentation (nixpkgs-fmt).
Or, shell completion is it really not working? I asked. No answer.
If you just do what I said (4 things), I think we are good to go.

superherointj commented on pkgs/applications/networking/cluster/cmctl/default.nix:
```diff
@@ -2,33 +2,49 @@
-    repo = "cert-manager";
-    rev = "v${version}";
-    sha256 = "sha256-sfC1acnCrcQ4A1tXXcjh47Af6xeJqjdGXy0gK21ZSFg=";
+    repo  = "cert-manager";
+    rev   = "v${version}";
+    hash  = "sha256-Z1aJ18X4mfJPlCPBC7QgfdX5Tk4+PK8mYoJZhGwz9ec=";
```
Nix doesn't use Go style indentation. Please remove the space. Or use a formatter (`nixpkgs-fmt`).

superherointj commented on pkgs/applications/networking/cluster/cmctl/default.nix:
```diff
@@ -2,33 +2,49 @@
-  ldflags = [ "-s" "-w" ];
+  ldflags = [
+    "-s" "-w"
+    # See https://github.com/cert-manager/cert-manager/blob/4486c01f726f17d2790a8a563ae6bc6e98465505/make/cmctl.mk#L1
+    "-X" "github.com/cert-manager/cert-manager/cmd/ctl/pkg/build.name=cmctl"
+    "-X" "github.com/cert-manager/cert-manager/cmd/ctl/pkg/build/commands.registerCompletion=true"
```
It's a single parameter, I'd prefer to have one per line.

superherointj commented on pkgs/applications/networking/cluster/cmctl/default.nix:
```diff
@@ -2,33 +2,49 @@
 
   nativeBuildInputs = [ installShellFiles ];
 
+  # Ensure we capture runtime completions by calling `completion` subcommand.
```
This comment is not necessary.

JoshVanL commented:
> I gave you feedback 4 times. I haven't heard from you on any of them.
For example, you are using Go style indentation instead of Nix indentation (nixpkgs-fmt).
Or, shell completion is it really not working? I asked. No answer. Note, that you are changing code that I'm not sure is the problem you pointed out. So I want to confirm that.
If you just do what I said (4 things), I think we are good to go.

Hi @superherointj, sorry- I can't see any of that review from my side. It might be a case that that the GitHub review is still pending and needs to be "submitted"?

superherointj commented on pkgs/applications/networking/cluster/cmctl/default.nix:
```diff
@@ -2,33 +2,49 @@
-      --bash <($out/bin/cmctl completion bash) \
-      --fish <($out/bin/cmctl completion fish) \
-      --zsh <($out/bin/cmctl completion zsh)
+    $out/bin/cmctl completion bash > cmctl.bash
+    $out/bin/cmctl completion zsh  > cmctl.zsh
+    $out/bin/cmctl completion fish > cmctl.fish
+    installShellCompletion cmctl.{bash,zsh,fish}
```
My expectation is the previous code to be equivalent to the code you are suggestion. I couldn't reproduce problem. So I'd like to confirm that this is really causing a problem before we change it.

JoshVanL replied:
So by "runtime completions" I mean CLI completions whereby suggestions are derived at runtime, rather than just the static sub-command suggestions. In `cmctl`s case, this is Kubernetes resources.

As an example, if I type the following command and press  `<TAB>`, I get suggestions for the current list of Namespaces:

```terminal
$ cmctl approve -n
another-namespace   cert-manager        default             kube-node-lease     kube-public         kube-system         local-path-storage  my-namespace
```

This functionality was missing before this PR, and pressing `<TAB>` wouldn't show anything.

JoshVanL replied:
Hmm, taking another look I understand your point now. I'm doing the same thing. 

I'll double check that it definitely wasn't working before again. I agree, your previous method is more concise.

JoshVanL replied:
Yes- sorry for the noise here. I must have gotten myself confused with another package..

I've reverted back the changes.

superherointj replied:
But what is causing this? Is really the way `installShellCompletion` is being executed?

JoshVanL replied:
No, everything was working correctly the way it was before. For whatever reason, I got it into my head that it needed changing and didn't read the previous command properly.

superherointj commented on pkgs/applications/networking/cluster/cmctl/default.nix:
```diff
@@ -2,37 +2,53 @@
     license = licenses.asl20;
     homepage = "https://cert-manager.io/";
-    maintainers = with maintainers; [ superherointj ];
+    maintainers = with maintainers; [ superherointj joshvanl ];
```
```suggestion
    maintainers = with maintainers; [ joshvanl superherointj ];
```
Ideally we should follow alphabetical order on lists. (Same applies to other fields.)

superherointj commented:
> Hi @superherointj, sorry- I can't see any of that review from my side. It might be a case that that the GitHub review is still pending and needs to be "submitted"?

Thanks for pointing out problem here. Sorry about that. On my POV you were just ignoring my comments. :D

JoshVanL commented:
>  Thanks for pointing out problem here. Sorry about that. On my POV you were just ignoring my comments. :D

No wozza! :slightly_smiling_face: 

I'll get those fixed up.

superherointj commented on pkgs/applications/networking/cluster/cmctl/default.nix:
```diff
@@ -2,20 +2,24 @@
-  ldflags = [ "-s" "-w" ];
+  ldflags = [
+    "-s" "-w"
+    "-X" "github.com/cert-manager/cert-manager/cmd/ctl/pkg/build.name=cmctl"
+    "-X" "github.com/cert-manager/cert-manager/cmd/ctl/pkg/build/commands.registerCompletion=true"
```
```suggestion
    "-X github.com/cert-manager/cert-manager/cmd/ctl/pkg/build.name=cmctl"
    "-X github.com/cert-manager/cert-manager/cmd/ctl/pkg/build/commands.registerCompletion=true"
```

JoshVanL commented:
@superherointj, hopefully fixed all of your review comments. PTAL

superherointj commented:
Result of `nixpkgs-review pr 183093` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cmctl</li>
  </ul>
</details>
